pkg/cache/cache_test.go

```diff
@@ -22,40 +22,96 @@ import (
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	kcorev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	kmetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kcache "k8s.io/client-go/tools/cache"
 
 	"sigs.k8s.io/controller-runtime/pkg/cache"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const testNamespaceOne = "test-namespace-1"
+const testNamespaceTwo = "test-namespace-2"
+
+func createPod(name, namespace string, restartPolicy kcorev1.RestartPolicy) runtime.Object {
```

Comment:
This seems generally useful to a variety of tests.  Add `TODO(community): ` to pull these into testenv.  Perhaps these should be functions on testenv.Environment.